lib/util/vulnerability/utils.ts

```diff
@@ -0,0 +1,30 @@
+export function setHighestVulnerabilitySeverity<
+  T extends Record<string, any>,
+  TChild extends Record<string, any> | undefined
+>(config: Record<string, any>, parent: T, child: TChild): void {
+  let mostSevere: string | undefined;
+
+  const parentVulSeverity = parent.vulnerabilitySeverity?.toUpperCase();
+
+  const childVulSeverity = child?.vulnerabilitySeverity?.toUpperCase();
+
+  if (!parentVulSeverity && childVulSeverity) {
+    mostSevere = childVulSeverity;
+  } else if (parentVulSeverity && !childVulSeverity) {
+    mostSevere = parentVulSeverity;
+  } else if (parentVulSeverity === 'CRITICAL') {
+    mostSevere = 'CRITICAL';
+  } else if (parentVulSeverity === 'HIGH' && childVulSeverity !== 'CRITICAL') {
+    mostSevere = 'HIGH';
+  } else if (
+    parentVulSeverity === 'MODERATE' &&
+    childVulSeverity !== 'CRITICAL' &&
+    childVulSeverity !== 'HIGH'
+  ) {
+    mostSevere = 'MODERATE';
+  } else {
+    mostSevere = childVulSeverity;
+  }
```

Comment:
This feels like an awkward way to do this. What about something [like this][1]

[1]: https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAcwKZQBI2QC1QZygDUQAbMVAJwEMAjGUmKATwGVUA3Kp5gHgChEiACqJUADyiowAE3yIASqghxKM3oUowwyADSJqYZgD5dgkQGEcDGWMnS5i5avWbteg0eOIAPonAyqMDaqDL8xgAUAA7UlNJQAFwi+hDWpDJJwlY2AJRJbjqIAN7mKmCEiDFxYMRk7FxaLIgAvJWx8QB0HGQUNPSMLPXcLAD8HVBwAKpRUVQW1PioETkA3KUIFak2JKRDjcwtiFvpY93kVHQMPHs8YxPTs5Tzi8tr6+VQiIsNPADyalQkkoVGoNFAtDp9GAQABbWhUbytEpCIQAGV+AHUkgBGMwoxAAWV+ABEAKIKACCwlJSQATHiURgAJIAcQwSQAzAyhBYFEzhEyLBTUUkACwMgC+byEMGAiAixxkOxuTWaav8siCIRkOWK5iEcSgIEoSCq8WVnGGzDWQgl-HMsvlZpqFp+qvVAS1FB1evxhuNpvaLrqlv2NsQdvM-pNX1DfwBlAA2s7ars4ywALqIYytb5W-6BJOK11WjP6xAjNrVVMq5jlpLFkNu638SNlCp5-aHNCYbB4Qg7XqXAZsdPMCLmIqIM5D-rXMdJABERLJlOpi4jDKnM4uc8GC8Qi+ZbI3dtW9vbcFIqA6pDgyAinZ456AA

Reply:
Absolutely, that's way more elegant, thanks @JamieMagee 🙇